source/aws_profile.c

```diff
@@ -895,6 +991,16 @@ static bool s_parse_profile_declaration(
             return true;
         }
 
+        s_parse_by_character_predicate(&profile_cursor, s_is_whitespace, NULL, 0);
+    } else if (has_sso_session_prefix) {
+        if (context->profile_collection->profile_source == AWS_PST_CREDENTIALS) {
```

Comment:
if we dont allow profiles or other section types in cred files, do we need to have that check inside of specific section type if

Reply:
Not sure how to improve this, as a profile can be default without any prefix. This is only invalid if there was a prefix other than profile, and the source is CREDENTIALS.